ddtrace/_trace/tracer.py

```diff
@@ -489,16 +480,15 @@ def configure(
             self._partial_flush_min_spans = partial_flush_min_spans
 
         if appsec_enabled is not None:
-            self._asm_enabled = asm_config._asm_enabled = appsec_enabled
+            asm_config._asm_enabled = appsec_enabled
 
         if iast_enabled is not None:
-            self._iast_enabled = asm_config._iast_enabled = iast_enabled
+            asm_config._iast_enabled = iast_enabled
 
         if appsec_standalone_enabled is not None:
-            self._appsec_standalone_enabled = asm_config._appsec_standalone_enabled = appsec_standalone_enabled
+            asm_config._appsec_standalone_enabled = appsec_standalone_enabled
 
-        if self._appsec_standalone_enabled and (self._asm_enabled or self._iast_enabled or config._sca_enabled):
-            self._apm_opt_out = True
+        if asm_config._apm_opt_out:
```

Comment:
I saw there's already ASM approval on the PR, but are these actually equivalent conditions (and this opt out isn't a consequence of these conditions being met)? Can we safely rely on the ASM config to update this for us, or should we just keep the same checks but on `asm_config.*` instead of `self.*`?